### ISSUE TITLE
Feature request: cutting demos

### ISSUE THREAD

**Issue by BerserkerBG** [stijn-volckaert/udemo#107]
Would be nice if it was possible to cut demos.

**Comment by SeriousBuggie** [comment]
What do you mean exactly? Like cut at the end? Because that only one operation which is possible. We can't trim it from the start for example.

**Comment by BerserkerBG** [comment]
Yes I talk about demo trimming. Well, if we can't trim demos from the start then that tool will not be very useful. Do you think its possible to fix demo recording driver so we can be able to trim new demos recorded with it (not compatible with old demos)?

**Comment by SeriousBuggie** [comment]
No. Demo is dump of raw net packets, sent from the server (recorder) to the client (player), via file.

There no some special protocol, separate protocol. It use same net relevancy and replication rules.

So at point X, you need know all info about what be before, so you can properly decode current packet and apply it on top of current state.

Best we can make here - is recording a demo from the demo.
But recording a demo is lossy operation. Also that not always goes for all mods, same as make online work not mean you make demoplay work.
So likely that demo will be broken. That need to check, but my expectation is low on that matter.

**Comment by SeriousBuggie** [comment]
I found interesting ability make this possible. Demo not be cut, but at end stored special info which understand by modern client, and then played as part from t1 to t2, when make seek from 0 to t1 on start. Still seek make huge delay on start play such demo. It is what you looking for?

**Comment by BerserkerBG** [comment]
The problem is seeking for longer demos. Seeking Siege demos for example is an absolute nightmare, there is a lot going on with Siege games and sometimes they last for hours. That makes seeking incredibly slow because actors accumulate until it clogs and seeking can sometimes take minutes, it will freeze your UT.

Honestly recording demo from demo is not a bad idea, if we play the demo with frame based playback, we can also slow down the demo play to make it more accurate and make sure there are no stutters or lags?

**Comment by SeriousBuggie** [comment]
It is bad idea. Play demo already screw things a lot. Attempt record from 3rd level of replication is nightmare even if possible at all.

If your real problem is alow seeking - you need make issue about that, not try invent solution, which is not related here.

Trim demos and slow seek is different goals. And we maybe can solve one without another. Anyway we need push to real goal.
